### PR TITLE
replaced use of subprocess with call_binary() where possible (idp_node)

### PR DESCRIPTION
**Pull request by William-Hill**

### CODE REVIEW

File: idp_node/myproxy.py [hunk 1]
```diff
@@ -231,13 +239,17 @@ def restart_myproxy_server():
...
-    if "running" in status["stdout"]:
-        return (True, status)
-    else:
+    try:
+        status = esg_functions.call_binary("/etc/init.d/myproxy-server", ["status"])
+    except ProcessExecutionError:
+        pass
+    try:
+        status = esg_functions.call_binary("/etc/init.d/myproxy", ["status"])
+    except ProcessExecutionError:
```
Comment (nathanlcarlson) [inline idp_node/myproxy.py]: Is there a reason these aren't of the form `service myproxy status`?

File: idp_node/myproxy.py [hunk 2]
```diff
@@ -208,19 +209,26 @@ def config_myproxy_server(globus_location, install_mode="install"):
...
-        print " MyProxy - Starting server..."
-        esg_functions.stream_subprocess_output("/etc/init.d/myproxy start")
+    try:
+        esg_functions.call_binary("/etc/init.d/myproxy-server", ["start"])
+    except ProcessExecutionError:
+        pass
+    try:
+        esg_functions.call_binary("/etc/init.d/myproxy", ["start"])
+    except ProcessExecutionError:
+        logger.error("Error starting Myproxy")
```
Comment (nathanlcarlson) [inline idp_node/myproxy.py]: Same comment as below, is there a reason these are not use `service`?

File: idp_node/globus.py [hunk 3]
```diff
@@ -243,22 +241,18 @@ def _install_globus(config_type):
...
     try:
-        esg_functions.stream_subprocess_output('''/usr/sbin/useradd -r -c "Globus System User" -g {globus_sys_acct_group} -p {globus_sys_acct_passwd} -s /bin/bash {globus_sys_acct}'''.format(globus_sys_acct_group="globus", globus_sys_acct_passwd=globus_sys_acct_passwd, globus_sys_acct=globus_sys_acct))
-    except SubprocessError, error:
-        logger.debug(error.__dict__["data"]["returncode"])
-        if error.__dict__["data"]["returncode"] == 9:
+        esg_functions.call_binary("useradd", useradd_options)
+    except ProcessExecutionError, err:
+        if err.retcode == 9:
             pass
+        else:
```
Comment (nathanlcarlson) [inline idp_node/globus.py]: I've seen this `useradd` pattern elsewhere. Could we have a `create_unix_user`, just like `create_unix_group`?

Comment (nathanlcarlson) [reply idp_node/globus.py]: Or simply `create_user`, `create_group`?

File: idp_node/myproxy.py (hunk 1 above)
Comment (nathanlcarlson) [reply idp_node/myproxy.py]: Okay, I did some reading on stackoverflow and it looks like we should use service to be more future-proof.

Comment (William-Hill) [reply idp_node/myproxy.py]: Just pushed out a change for it

**Pull request merged**